dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/processing/CommonQueryRequestProcessor.java

```diff
@@ -95,4 +129,107 @@ private CommonQueryRequest computePagingParams( CommonQueryRequest commonQueryRe
+     *
+     * @param commonQueryRequest the {@link CommonQueryRequest} to transform
+     * @return the transformed {@link CommonQueryRequest}
+     */
+    private CommonQueryRequest computeProgramStatus( CommonQueryRequest commonQueryRequest )
+    {
+        if ( commonQueryRequest.hasProgramStatus() || commonQueryRequest.hasEnrollmentStatus() )
+        {
+            // merging programStatus and enrollmentStatus into a single set
+            Set<String> enrollmentStatuses = new LinkedHashSet<>();
+            enrollmentStatuses.addAll( commonQueryRequest.getProgramStatus() );
+            enrollmentStatuses.addAll( commonQueryRequest.getEnrollmentStatus() );
+
+            commonQueryRequest.getDimension()
+                .addAll( programStatusAsDimension( enrollmentStatuses ) );
+        }
+        return commonQueryRequest;
+    }
+
+    /**
+     * Converts the event status into a static dimension example:
+     * eventStatus=IpHINAT79UW.A03MvHHogjR.SCHEDULE;IpHINAT79UW.A03MvHHogjR.ACTIVE
+     * becomes dimension=IpHINAT79UW.A03MvHHogjR.EVENT_STATUS:SCHEDULE;ACTIVE
+     *
+     * @param commonQueryRequest the {@link CommonQueryRequest} to transform
+     * @return the transformed {@link CommonQueryRequest}
+     */
+    private CommonQueryRequest computeEventStatus( CommonQueryRequest commonQueryRequest )
+    {
+        if ( commonQueryRequest.hasEventStatus() )
+        {
+            commonQueryRequest.getDimension()
+                .addAll( eventStatusAsDimension( commonQueryRequest.getEventStatus() ) );
+        }
+        return commonQueryRequest;
+    }
+
+    private Collection<String> eventStatusAsDimension( Set<String> eventStatuses )
+    {
+        // builds a map of [program,program stage] with a list of event statuses
+        Map<Pair<String, String>, List<EventStatus>> statusesByProgramAndProgramStage = eventStatuses.stream()
+            .map( eventStatus -> splitAndValidate( eventStatus, 3, E7141 ) )
+            .map( parts -> Pair.of( Pair.of( parts[0], parts[1] ), EventStatus.valueOf( parts[2] ) ) )
+            .collect( groupingBy( Pair::getLeft, mapping( Pair::getRight, Collectors.toList() ) ) );
+
+        return statusesByProgramAndProgramStage.keySet()
+            .stream()
+            .map( programWithStage -> programWithStage.getLeft() + // program
+                DIMENSION_IDENTIFIER_SEP +
+                programWithStage.getRight() + // program stage
+                DIMENSION_IDENTIFIER_SEP +
+                EVENT_STATUS.name() + // "EVENT_STATUS"
+                DIMENSION_NAME_SEP +
+                // ";" concatenated values - for example "COMPLETED;SKIPPED"
+                statusesByProgramAndProgramStage.get( programWithStage ).stream()
+                    .map( EventStatus::name )
+                    .collect( Collectors.joining( ";" ) ) )
+            .collect( Collectors.toList() );
+    }
+
+    private List<String> programStatusAsDimension( Set<String> programStatuses )
```

Comment:
I believe would be better to use enrollment instead of program.
So, `enrollmentStatusAsDimension` as dimension instead of `programStatusAsDimension`

Reply:
Also, could we use the enum `EnrollmentStatus` instead of `ProgramStatus`?